src/compiler/parser.ts

```diff
@@ -3039,6 +3076,10 @@ namespace ts {
         }
 
         function parseAssignmentExpressionOrHigher(): Expression {
+            return doInExpressionContext(parseAssignmentExpressionOrHigherNoContextSet);

```

Comment:
In most cases you shouldn't need to set context before parsing an assignment expression, so you could call `parseAssignmentExpressionOrHigherNoContextSet` directly.

Reply:
Done